Subscription/Subscription.sol

```diff
@@ -240,52 +243,29 @@ contract Subscription {
         returns (bool success)
     {
         // make sure the subscription is valid and ready
-        // pulled this out so I have the hash, should be exact code as "isSubscriptionReady"
-        bytes32 subscriptionHash = getSubscriptionHash(
-            from, to, tokenAddress, tokenAmount, periodSeconds, gasPrice, nonce
-        );
-        address signer = getSubscriptionSigner(subscriptionHash, signature);
+        require(isSubscriptionReady(to, from, tokenAddress, tokenAmount, periodSeconds, gasPrice, nonce, signature), "Subscription is not ready or conditions of transction are not met")
 
-        //make sure they aren't sending to themselves
-        require(to != from, "Can not send to the from address");
-        //the signature must be valid
-        require(signer == from, "Invalid Signature");
-        //timestamp must be equal to or past the next period
+        // now, let make the transfer from the subscriber to the publisher
+        uint256 startingBalance = ERC20(tokenAddress).balanceOf(to);
+        ERC20(tokenAddress).transferFrom(from,to,tokenAmount);
         require(
-            block.timestamp >= nextValidTimestamp[subscriptionHash],
-            "Subscription is not ready"
+          (startingBalance+tokenAmount) == ERC20(tokenAddress).balanceOf(to),
+          "ERC20 Balance did not change correctly"
         );
 
-        // if there are requirements from the deployer, let's make sure
-        // those are met exactly
-        require( requiredToAddress == address(0) || to == requiredToAddress );
-        require( requiredTokenAddress == address(0) || tokenAddress == requiredTokenAddress );
-        require( requiredTokenAmount == 0 || tokenAmount == requiredTokenAmount );
-        require( requiredPeriodSeconds == 0 || periodSeconds == requiredPeriodSeconds );
-        require( requiredGasPrice == 0 || gasPrice == requiredGasPrice );
```

Comment:
4. 

removed the numbered checks above because they are called within `isSubscriptionReady` or within `getSubscriptionHash` which is called within `isSubscriptionReady`. 

The only down side to relying on `isSubscriptionReady` is that we lose some fidelity in error messages. But we gain DRY (Dont Repeat Yourself) code.